src/test/ui/dyn-keyword/dyn-2021-edition-error.rs

```diff
@@ -0,0 +1,12 @@
+// edition:2021
```

Comment:
Do we have tests for this for edition 2018? Just want to make sure we test for the applicable suggestions not to break. Also, I don't think we've had a discussion around test coverage for people updating from editions earlier than CURRENT - 1. Should we be having that conversation?

Reply:
My understanding is that that's simply not supported, you have to upgrade editions one by one.

Reply:
So, we are currently testing Rust2018 though virtue of the compiler being on Rust 2018 itself. However, once it moves to Rust 2021 (no idea when that [will happen](https://rust-lang.zulipchat.com/#narrow/stream/268952-edition-2021/topic/Updating.20rustc.20to.20use.20Rust.202021)), the tests that currently only assume the current edition ([like this one](https://github.com/rust-lang/rust/blob/673d0db5e393e9c64897005b470bfeb6d5aec61b/src/test/ui/lint/bare-trait-objects-path.rs)) will need to be updated. 

We could explicitly make a test right now, making sure that the lint triggers in 2018 which will be a bit of a duplicate test but will ensure we still test this code path when the compiler is updated.